homeassistant/components/motioneye/__init__.py

```diff
@@ -0,0 +1,231 @@
+"""The motionEye integration."""
+from __future__ import annotations
+
+import asyncio
+import logging
+from typing import Any, Callable
+
+from motioneye_client.client import (
+    MotionEyeClient,
+    MotionEyeClientError,
+    MotionEyeClientInvalidAuth,
+)
+from motioneye_client.const import KEY_CAMERAS, KEY_ID, KEY_NAME
+
+from homeassistant.config_entries import SOURCE_REAUTH, ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_PORT, CONF_SOURCE
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.dispatcher import (
+    async_dispatcher_connect,
+    async_dispatcher_send,
+)
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import (
+    CONF_ADMIN_PASSWORD,
+    CONF_ADMIN_USERNAME,
+    CONF_CLIENT,
+    CONF_COORDINATOR,
+    CONF_ON_UNLOAD,
+    CONF_SURVEILLANCE_PASSWORD,
+    CONF_SURVEILLANCE_USERNAME,
+    DEFAULT_SCAN_INTERVAL,
+    DOMAIN,
+    SIGNAL_CAMERA_ADD,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = ["camera"]
+
+
+def create_motioneye_client(
+    *args: Any,
+    **kwargs: Any,
+) -> MotionEyeClient:
+    """Create a MotionEyeClient."""
+    return MotionEyeClient(*args, **kwargs)
+
+
+def get_motioneye_config_unique_id(host: str, port: int) -> str:
+    """Get the unique_id for a motionEye config."""
+    return f"{host}:{port}"
```

Comment:
this does not comply https://developers.home-assistant.io/docs/entity_registry_index/#unacceptable-sources-for-a-unique-id

Reply:
I agree, it does not. There was a [very brief discord chat](https://discord.com/channels/330944238910963714/554842238073700352/825391070846255154) on this. In summary, I have nothing else available. It's either host:port, or the config entry id. 

Whilst I am happy to change it to the config entry id, I don't understand why that is better. As it stands, at least basic non-malicious duplication will be prevented. If I use config entry id, I can literally recreate multiple devices pointed a the same host:port.

Thoughts welcome.

Reply:
Do not add a unique ID if you have none. Config Entry ID is not going to help, it's only for deduplication. You can still do host deduplication but you need to iterate over the config entries yourself instead of using unique ID.

Reply:
Thank you for the authoritative answer @balloob ! 

After the original feedback from @dgomes , I took the liberty of stripping the unique_id entirely out of the config flow -- so I think what is here is now compatible with your suggestion. Note that I still use the config entry ID [as part of the unique_id for entities](https://github.com/home-assistant/core/pull/48239/files#diff-3ca7aab36b02ba61a5678566eae81031138bf82c42a2866a36a95523d1259ba7R60) .

Reply:
Perfect 👍 and yes using config entry ID is great for entity unique ID.